### PR TITLE
power source recording,playback and import working fine

### LINKED ISSUE #1861: Record feature for power source not working properly
**Actual Behaviour**
Currently the record feature is not working as expected. File is not getting generated

**Expected Behaviour**
Recording should work properly.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by neel1998**
Fixes #1861

**Changes**: User can log, playback and import already saved log in power source

**Screenshot/s for the changes**: 

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[power_source.zip](https://github.com/fossasia/pslab-android/files/3386177/power_source.zip)

### CODE REVIEW

**Comment by CloudyPadmal** [comment]
I don't see it recording. It's like taking a snapshot of values and saving.

**Comment by neel1998** [comment]
@CloudyPadmal  can you please have a look at this

**Comment by cweitat** [comment]
Conflicts now

**Comment by neel1998** [comment]
@cweitat yes. I will fix conflicts today.

**Comment by neel1998** [comment]
@CloudyPadmal @cweitat  please review and merge

File: app/src/main/java/io/pslab/interfaces/sensorloggers/PowerSourceRecordables.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package io.pslab.interfaces.sensorloggers;
+
+import io.pslab.models.PowerSourceData;
+import io.realm.RealmResults;
+
+public interface PowerSourceRecordables {
+    PowerSourceData getPowerData(long timeStamp);
+
```
Comment (cweitat) [inline app/src/main/java/io/pslab/interfaces/sensorloggers/PowerSourceRecordables.java]: Excessive line breaks?

Comment (neel1998) [reply app/src/main/java/io/pslab/interfaces/sensorloggers/PowerSourceRecordables.java]: There are many other classes wth same problem of excessive line breaks. I think i should make a separate clean up PR. will it be okay?

Comment (CloudyPadmal) [reply app/src/main/java/io/pslab/interfaces/sensorloggers/PowerSourceRecordables.java]: For this file do here

Comment (neel1998) [reply app/src/main/java/io/pslab/interfaces/sensorloggers/PowerSourceRecordables.java]: Done

**Comment by neel1998** [comment]
@CloudyPadmal conflicts resolved. Please merge

**Comment by neel1998** [comment]
@CloudyPadmal  @cweitat  please merge this

**Pull request merged**